tests/RobotFramework/tests/cumulocity/self-update/tedge_self_update.robot

```diff
@@ -74,6 +73,25 @@ Refreshes mosquitto bridge configuration
     # Mosquitto should be restarted with new bridge
     Execute Command    cmd=sh -c '[ $(journalctl -u mosquitto | grep -c "Loading config file /etc/tedge/mosquitto-conf/c8y-bridge.conf") = 2 ]'
 
+Update tedge version from base to current using Cumulocity
+    ${arch}=    Get Debian Architecture
+    Execute Command    mkdir -p /setup/base-version
+    Transfer To Device    ${CURDIR}/base-version/*${arch}.deb    /setup/base-version/
+
+    # Install base version with self-update capability
+    Install Packages    /setup/base-version
+    Execute Command    cd /setup && test -f ./bootstrap.sh && ./bootstrap.sh --no-install --no-secure
+    Device Should Exist                      ${DEVICE_SN}
+    ${pid_before}=    Execute Command    pgrep tedge-agent    strip=${True}
+
+    # Upgrade to current version
+    Create Local Repository
+    ${OPERATION}=    Install Software    tedge,${NEW_VERSION}    tedge-mapper,${NEW_VERSION}    tedge-agent,${NEW_VERSION}    tedge-watchdog,${NEW_VERSION}    tedge-apt-plugin,${NEW_VERSION}
+
+    Operation Should Be SUCCESSFUL    ${OPERATION}    timeout=60
+    ${pid_after}=    Execute Command    pgrep tedge-agent    strip=${True}
+    Should Not Be Equal    ${pid_before}    ${pid_after}
+
```

Comment:
The other reason why it was omitted was because doing the on-disk binary version check was not sufficient, as this issue itself was about the running version being different from the one on the disk. But, as an upgrade test, it makes sense to keep them both to make the test more complete.